pkg/engine/executor/pipeline.go

```diff
@@ -0,0 +1,122 @@
+package executor
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/apache/arrow-go/v18/arrow"
+)
+
+type Transport uint8
+
+const (
+	_ Transport = iota
+	Local
+	Remote
+)
+
+// Pipeline represents a data processing pipeline that can read Arrow records.
+// It provides methods to read data, access the current record, and close resources.
+type Pipeline interface {
+	// Read reads the next value into its state.
+	// It returns an error if reading fails or when the pipeline is exhausted. In this case, the function returns EOF.
+	// The implementation must retain the returned error in its state and return it with subsequent Value() calls.
+	Read() error
```

Comment:
just curious to know why reads are split across two calls, Read() followed by Value()

Reply:
Initially I thought about passing the number of desired rows to the `Read() function and return the actual number of rows, but that is kinda redundant to the batch size.

```go
type Pipeline interface {
    Read(int64) (int64, error)
    Value() (state)
}
```